### PR TITLE
Supertask table

### PR DESCRIPTION
**Pull request by xbenyx**
This pull request adds the Supertask table. 

Currently is missing the Modal Subtasks, but leaving this for discussion, to see if its better Modal or Subtable.

### CODE REVIEW

File: src/app/core/_components/tables/supertasks-pretasks-table/supertasks-pretasks-table.component.ts [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+          })
+        )
+        .subscribe(() => {
+          this.snackBar.open('Successfully deleted pretasks!', 'Close');
+          this.reload();
+        })
+    );
+  }
+
+  renderPriorityColumn(pretask: Pretask): SafeHtml {
```
Comment (itbf9) [inline src/app/core/_components/tables/supertasks-pretasks-table/supertasks-pretasks-table.component.ts:289]: You can look at the tasks-table how to implement an editable field

File: src/app/core/_components/tables/supertasks-table/supertasks-table.component.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+            `Successfully deleted ${results.length} supertasks!`,
+            'Close'
+          );
+          this.reload();
+        })
+    );
+  }
+
+  @Cacheable(['_id'])
+  async renderSupertaskLink(
```
Comment (itbf9) [inline src/app/core/_components/tables/supertasks-table/supertasks-table.component.ts:216]: Maybe this one can be declared in the base class with the other links?

File: src/app/core/_components/tables/supertasks-table/supertasks-table.component.ts [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    const dialogRef = this.dialog.open(ModalPretasksComponent, {
+      width: '100%',
+      data: {
+        supertaskId: supertask._id,
+        supertaskName: supertask.supertaskName
+      }
+    });
+
+    dialogRef.afterClosed().subscribe((result) => {
+      console.log('The modal was closed');
```
Comment (itbf9) [inline src/app/core/_components/tables/supertasks-table/supertasks-table.component.ts:260]: debug output?

File: src/app/tasks/supertasks/modal-pretasks/modal-pretasks.component.html [hunk 4]
```diff
@@ -1,46 +1,7 @@
...
-  <button type="submit" class="btn btn-primary" (click)="onSubmit()">Update</button>
-  <button class="btn" style="background-color: rgb(138, 133, 132);" (click)="modal.close('Cancel')">Cancel</button>
-</div>
+<app-table>
+  <app-page-title [title]="supertaskName"></app-page-title>
+  <supertasks-pretasks-table [supertaskId]="supertaskId"></supertasks-pretasks-table>
+</app-table>
 
+
+<!-- <input type="number" class="form-control" id="Priority" formControlName="priority" placeholder="{{ pt.priority }}"> -->
```
Comment (itbf9) [inline src/app/tasks/supertasks/modal-pretasks/modal-pretasks.component.html:7]: remove

**Review by itbf9 — APPROVED** [review]
Some minor comments but i approve so that you can marge later.

File: src/app/core/_components/tables/supertasks-table/supertasks-table.component.ts (hunk 2 above)
Comment (xbenyx) [reply src/app/core/_components/tables/supertasks-table/supertasks-table.component.ts:211]: Yes, makes sense, completely missed that. thanks

**Pull request merged**